api/envoy/config/route/v3/scoped_route.proto

```diff
@@ -110,10 +116,17 @@ message ScopedRouteConfiguration {
   // The name assigned to the routing scope.
   string name = 1 [(validate.rules).string = {min_len: 1}];
 
-  // The resource name to use for a :ref:`envoy_v3_api_msg_service.discovery.v3.DiscoveryRequest` to an
-  // RDS server to fetch the :ref:`envoy_v3_api_msg_config.route.v3.RouteConfiguration` associated
-  // with this scope.
-  string route_configuration_name = 2 [(validate.rules).string = {min_len: 1}];
+  oneof route_config {
+    option (validate.required) = true;
+
+    // The resource name to use for a :ref:`envoy_v3_api_msg_service.discovery.v3.DiscoveryRequest` to an
+    // RDS server to fetch the :ref:`envoy_v3_api_msg_config.route.v3.RouteConfiguration` associated
+    // with this scope.
+    string route_configuration_name = 2 [(validate.rules).string = {min_len: 1}];
```

Comment:
Moving an existing field into a `oneof` is a breaking change, since it changes the API of the generated code in golang.  Instead, you'll need to keep this existing field outside of the `oneof` and add a `(udpa.annotations.field_migrate).oneof_promotion` option telling our tooling that the field should be moved into the oneof in the next major API version.

Reply:
Thanks for catching this; I've removed the `oneof` and used the UDPA annotation instead.